### PR TITLE
Improve Rhai errors

### PR DESCRIPTION
**Pull request by dbanty**
This PR attempts to improve the Rhai error handling and debugging experience. The only tweak to the code is that if there is an error in the main script, the path to that script is included in the error. Previously, the plugin name `apollo.rhai` (which _looks_ like a file name for a script) combined with the fact that line numbers/positions are incorrect in syntax errors, made it seem like some internal generated code was failing.

This does _not_ add paths to errors in imported modules, since the module names seemed clear enough to me.

Additionally, I added a "Debugging" section to the Rhai docs—this needs some additions from folks who know Rhai better than I do, but it has some basics to get started. Specifically, I think we should add some advice on testing Rhai scripts.

**Checklist**

Complete the checklist (and note appropriate exceptions) before a final PR is raised.

- [x] Changes are compatible[^1]
- [x] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [x] Unit Tests
    - [x] Integration Tests
    - [x] Manual Tests

### CODE REVIEW

File: apollo-router/src/plugins/rhai/mod.rs [hunk 1]
```diff
@@ -9,6 +9,7 @@ use std::sync::Arc;
 use std::sync::Mutex;
 use std::time::Duration;
 
+use anyhow::anyhow;
```
Comment (garypen) [inline apollo-router/src/plugins/rhai/mod.rs:12]: Don't need `anyhow`. `BoxError` will happily box up a `String`.

File: apollo-router/src/plugins/rhai/mod.rs [hunk 2]
```diff
@@ -81,7 +82,9 @@ impl EngineBlock {
             sdl.to_string(),
             main.clone(),
         ));
-        let ast = engine.compile_file(main)?;
+        let ast = engine
+            .compile_file(main.clone())
+            .map_err(|err| anyhow!("in Rhai script {}: {}", main.display(), err))?;
```
Comment (garypen) [inline apollo-router/src/plugins/rhai/mod.rs:87]:
```suggestion
            .map_err(|err| format!("in Rhai script {}: {}", main.display(), err))?;
```

**Review by garypen — CHANGES_REQUESTED** [review]
Thanks for the contribution. This is a nice change and should make life simpler for users.

I'll re-review after you look at my suggested changes.

**Comment by dbanty** [comment]
Thanks @garypen, all set! Can't believe I didn't know you could `From<String> for Box<dyn Error>`!

**Review by garypen — APPROVED** [review]

**Pull request merged**